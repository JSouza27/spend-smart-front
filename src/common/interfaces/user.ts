export interface IUser {
  id: string;
  name: string;
  email: string;
  imageUrl: null;
  additional_user_id: null;
  account: string;
  nick: string;
}
