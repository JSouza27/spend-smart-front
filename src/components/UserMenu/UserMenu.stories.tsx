import { ComponentMeta, Story } from '@storybook/react'
import { ThemeProvider } from 'styled-components';

import UserMenu from '.'
import theme from '../../styles/theme';

export default {
  title: 'UserMenu',
  component: UserMenu
} as ComponentMeta

export const Basic: Story = () => (
  <ThemeProvider theme={theme}>
    <UserMenu />
  </ThemeProvider>
)
