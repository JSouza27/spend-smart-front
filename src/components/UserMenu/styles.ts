import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  gap: 1rem;
`;

export const UserContainer = styled.div`
  ${({ theme }) => css`
    align-items: flex-start;
    display: flex;
    flex-direction: column;
    justify-content: center;

    span {
      color: ${theme.colors.neutral_800};
      font-size: ${theme.font.sizes.small};
      line-height: 2.4rem;
    }
  `}
`;

export const Avatar = styled.div`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.neutral_400};
    border: none;
    border-radius: 100%;
    color: ${theme.colors.neutral_700};
    display: flex;
    justify-content: center;
    height: 5rem;
    overflow: hidden;
    width: 5rem;
  `}
`;
