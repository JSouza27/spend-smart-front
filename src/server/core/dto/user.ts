export interface IUser {
  id: string;
  name: string;
  email: string;
  imageUrl: string | null;
  additional_user_id: string | null;
  cpf?: string;
  telefone?: string;
  account: string;
  nick: string;
}
