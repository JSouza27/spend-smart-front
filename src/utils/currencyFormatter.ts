export class CurrencyFormatter {
  private static language = 'pt-BR';
  private static currency = 'BRL';

  static formatter(num: number): string {
    return (num ?? 0).toLocaleString(this.language, {
      currency: this.currency
    });
  }

  static unformat(num: string): number {
    const nums = num.replace(/[^0-9]+/g, '');
    const i = nums.length - 2;

    return Number(`${nums.substring(0, i)}.${nums.substring(i)}`);
  }
}
