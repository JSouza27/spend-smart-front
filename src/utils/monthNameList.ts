export const monthNameList = [
  'janeiro',
  'fevereiro',
  'mar√ßo',
  'abril',
  'maio',
  'junho',
  'julho',
  'agosto',
  'setembro',
  'outubro',
  'novembro',
  'dezembro'
];
