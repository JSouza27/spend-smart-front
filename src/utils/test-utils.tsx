import { RenderOptions, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../styles/theme';

const AllTheProviders = ({ children }: any) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
