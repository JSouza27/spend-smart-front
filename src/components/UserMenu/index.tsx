import Image from 'next/image';
import Link from 'next/link';
import { DocumentTextExtract } from '@styled-icons/fluentui-system-filled';
import { User, Exit } from '@styled-icons/boxicons-regular';

import Heading from '../Heading';
import user from 'data/constants/mockUsuario';
import * as S from './styles';
import Dropdown from '../Dropdown';

function Trigger() {
  return (
    <S.Wrapper>
      <S.UserContainer>
        <Heading level={2} size="medium" lineHeight="2.4rem">
          {user.name}
        </Heading>
        <span>{user.email}</span>
      </S.UserContainer>
      <S.Avatar>
        {user.imageUrl ? (
          <Image
            src={user.imageUrl}
            width={500}
            height={500}
            alt="Picture of the author"
          />
        ) : (
          <Heading
            level={1}
            size="xlarge"
            color="neutral_800"
            lineHeight="2.4rem"
          >
            {user.nick}
          </Heading>
        )}
      </S.Avatar>
    </S.Wrapper>
  );
}

export default function UserMenu() {
  return (
    <Dropdown trigger={<Trigger />}>
      <S.Nav>
        <S.Title>
          <Heading
            level={1}
            size="xsmall"
            color="neutral_600"
            lineHeight={'1.6rem'}
          >
            Usu√°rio
          </Heading>
        </S.Title>
        <Link href="/extrato" passHref>
          <S.Link>
            <DocumentTextExtract size={16} />
            <span>Extrato</span>
          </S.Link>
        </Link>
        <Link href="/profile" passHref>
          <S.Link>
            <User size={16} />
            <span>Meus Dados</span>
          </S.Link>
        </Link>
        <Link href="/logout" passHref>
          <S.Link>
            <Exit size={16} />
            <span>Sair</span>
          </S.Link>
        </Link>
      </S.Nav>
    </Dropdown>
  );
}
