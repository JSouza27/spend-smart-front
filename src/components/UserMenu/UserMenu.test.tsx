import { render, screen } from 'utils/test-utils';

import UserMenu from '.';
import user from 'data/mocks/mockUsuario';

describe('<UserMenu />', () => {
  it('should render the heading', () => {
    const { container } = render(<UserMenu />);

    expect(
      screen.getByRole('heading', { name: user.name })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
