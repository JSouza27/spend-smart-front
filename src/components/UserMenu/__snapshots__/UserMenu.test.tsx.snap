// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenu /> should render the heading 1`] = `
.c2 {
  color: neutral_900;
  font-size: xlarge;
  font-weight: 600;
  line-height: 2.4rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 span {
  color: #474D66;
  line-height: 2.4rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #D8DAE5;
  border: none;
  border-radius: 100%;
  color: #696F8C;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5rem;
  overflow: hidden;
  width: 5rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Jo√£o da Silva
    </h2>
    <span>
      jjjjoao@xmail.com
    </span>
  </div>
  <div
    class="c3"
  >
    <h1
      class="c2"
    >
      JS
    </h1>
  </div>
</div>
`;
